modules/lmdbbackend/lmdbbackend.cc

```diff
@@ -2014,6 +2099,46 @@ void LMDBBackend::lookupStart(domainid_t domain_id, const std::string& match, bo
 
 bool LMDBBackend::getInternal(DNSName& basename, std::string_view& key)
 {
+  // FIXME: bit of duplication from below here, but much simpler
+  if (d_lookupstate.comments) {
+    if (!d_lookupstate.cursor) {
+      d_rotxn.reset();
+      return false;
+    }
+
+    key = d_lookupstate.key.getNoStripHeader<string_view>();
+    // remove hash from the key so compoundOrdername::get* work
+    key = key.substr(0, key.size() - 256 / 8);
+
+    basename = compoundOrdername::getQName(key);
+
+    uint64_t ts;
+    auto val = d_lookupstate.val.get<string_view>();
+    if (val.size() < sizeof(uint64_t)) {
```

Comment:
actually
```suggestion
    if (val.size() <= sizeof(uint64_t)) {
```
because you need at least a `NUL` terminator after the timestamp.